cpp/include/cudf/io/datasource.hpp

```diff
@@ -340,6 +369,9 @@ class arrow_io_source : public datasource {
   }
 
  private:
+  const std::string uri_start_delimiter = "//";
+  const std::string uri_end_delimiter   = "?";
```

Comment:
If these are only used in the ctor, I think it'd be better to make these local variables in the ctor.